helpers/datastructures.go

```diff
@@ -24,7 +27,7 @@ type InterfaceObj struct {
 
 func Error(e error) *ErrorObj                         { return &ErrorObj{key: "error", value: e} }
 func Int(k string, v int) *IntObj                     { return &IntObj{key: k, value: v} }
-func String(k, v string) *StringObj                   { return &StringObj{key: k, value: v} }
+func String(k, v string) *StringObj                   { return &StringObj{key: k, value: strings.ToValidUTF8(v, "")} }
```

Comment:
issue: replacing with an empty silently loses information

If we are trying to log an interpolated string with contents of an array or a map, and we use an empty string as a replacement, we might lose important information, like the fact that an element exists.

We should be especially concerned with this for remote logging, because when debugging a remote issue we can’t necessarily reproduce, if we lose out a valuable hint that a string contains an invalid UTF-8, we might waste our time on other leads or just give up reproducing.

Reply:
suggestion: we should agree on a replacement value for such cases

For example, `INVALID_UTF8_SEQ` might do? We could even use a single character like the Unicode Replacement Character:
> U+FFFD � REPLACEMENT CHARACTER used to replace an unknown, unrecognized, or unrepresentable character